WalletWasabi.Fluent/Models/Wallets/TransactionModelExtensions.cs

```diff
@@ -0,0 +1,21 @@
+using System.Linq;
+using NBitcoin;
+using WalletWasabi.Blockchain.Transactions.Summary;
+
+namespace WalletWasabi.Fluent.Models.Wallets;
+
+public static class TransactionModelExtensions
+{
+	public static Money OutputAmount(this ITransactionModel transaction) => transaction.Outputs.Sum(x => x.Amount);
+	public static Money? InputAmount(this ITransactionModel transaction) => transaction.Inputs.OfType<UnknownInput>().Any() ? null : transaction.Inputs.Cast<InputAmount>().Sum(x => x.Amount);
+
+	public static Money? Fee(this ITransactionModel transaction)
+	{
+		if (transaction.InputAmount() is { } amount)
+		{
+			return amount - transaction.OutputAmount();
+		}
+
+		return null;
+	}
```

Comment:
IMO the transaction fee should be calculated in `TransactionSummary.cs` class as all the other data.

And we don't have to introduce a class on UI side that deals with inputs and outputs.

Reply:
Please, check if things are better now :)